x-pack/plugins/security_solution/public/detections/components/endpoint_responder/use_responder_action_data.ts

```diff
@@ -6,53 +6,104 @@
  */
 import type { ReactNode } from 'react';
 import { useCallback, useMemo } from 'react';
-import { useGetEndpointDetails, useWithShowEndpointResponder } from '../../../management/hooks';
+import type { TimelineEventsDetailsItem } from '@kbn/timelines-plugin/common';
+import { getSentinelOneAgentId } from '../../../common/utils/sentinelone_alert_check';
+import type { ThirdPartyAgentInfo } from '../../../../common/types';
+import type { ResponseActionAgentType } from '../../../../common/endpoint/service/response_actions/constants';
+import { useGetEndpointDetails, useWithShowResponder } from '../../../management/hooks';
 import { HostStatus } from '../../../../common/endpoint/types';
 import {
-  NOT_FROM_ENDPOINT_HOST_TOOLTIP,
-  LOADING_ENDPOINT_DATA_TOOLTIP,
   HOST_ENDPOINT_UNENROLLED_TOOLTIP,
+  LOADING_ENDPOINT_DATA_TOOLTIP,
   METADATA_API_ERROR_TOOLTIP,
+  NOT_FROM_ENDPOINT_HOST_TOOLTIP,
 } from './translations';
+import { getFieldValue } from '../host_isolation/helpers';
 
 export interface ResponderContextMenuItemProps {
   endpointId: string;
   onClick?: () => void;
+  agentType?: ResponseActionAgentType;
```

Comment:
Not for this PR - but I would like to, at some point, to move all endpoint mangament type of code (component, hooks, etc) to a more centralized folder (probably under `public/common/endpoint` to the current `public/common/components/endpoint`).